install.hs

```diff
@@ -435,33 +435,43 @@ allVersionMessage wordList = case wordList of
     in  concat $ (init $ init msg) ++ [" and ", lastVersion]
 
 
--- TODO: more sophisticated interface to stack and cabal
+-- RUN EXECUTABLES
 
-execStackWithYaml_ :: VersionNumber -> [String] -> Action ()
-execStackWithYaml_ versionNumber args = do
+-- |Execute a stack command for a specified ghc, discarding the output
+execStackWithGhc_ :: VersionNumber -> [String] -> Action ()
+execStackWithGhc_ versionNumber args = do
   let stackFile = "stack-" ++ versionNumber ++ ".yaml"
   command_ [] "stack" (("--stack-yaml=" ++ stackFile) : args)
 
-execStackWithYaml :: CmdResult r => VersionNumber -> [String] -> Action r
-execStackWithYaml versionNumber args = do
+-- |Execute a stack command for a specified ghc
+execStackWithGhc :: CmdResult r => VersionNumber -> [String] -> Action r
+execStackWithGhc versionNumber args = do
   let stackFile = "stack-" ++ versionNumber ++ ".yaml"
   command [] "stack" (("--stack-yaml=" ++ stackFile) : args)
 
-execStack :: CmdResult r => [String] -> Action r
-execStack = command [] "stack"
+-- |Execute a stack command with the same resolver as the build script
+execStackShake :: CmdResult r => [String] -> Action r
+execStackShake args =
+  command [] "stack" ("--stack-yaml=shake.yaml" : args)
 
-execStack_ :: [String] -> Action ()
-execStack_ = command_ [] "stack"
+-- |Execute a stack command with the same resolver as the build script, discarding the output
+execStackShake_ :: [String] -> Action ()
+execStackShake_ args =
+  command_ [] "stack" ("--stack-yaml=shake.yaml" : args)
 
 execCabal :: CmdResult r => [String] -> Action r
-execCabal = command [] "cabal"
+execCabal =
+  command [] "cabal"
 
 execCabal_ :: [String] -> Action ()
 execCabal_ = command_ [] "cabal"
 
 existsExecutable :: MonadIO m => String -> m Bool
 existsExecutable executable = liftIO $ isJust <$> findExecutable executable
 
+
+-- QUERY ENVIRONMENT
```

Comment:
This could be more prominent, otherwise it can easily be overlooked.

Reply:
Suggestions?

Reply:
```haskell
-- ------------------------------------------------------------
-- * query environment
-- ------------------------------------------------------------
```
Have seen this in the cabal code base.
However, this is highly subjective suggestion, no need to do it.